docs/extension_build_system.rst

```diff
@@ -0,0 +1,31 @@
+======================
+Extension Build System
+======================
+
+
+Introduction
+------------
+
+By default, scikit-build looks in the project top-level directory for a
+file named ``CMakeLists.txt``. It will then invoke CMake executable specifying
+a generator matching the python being used.
+
+Indeed, each python version is associated with an official compiler. By default,
+scikit-build will automatically select the generator matching the official
```

Comment:
generator -> compiler, associated C runtime, and build flags

I think these or similar are more specific. Also, hopefully in the future we can use the Ninja generator.